### PR TITLE
Add cgroup v1 freezer controller

### PR DESCRIPTION
**Pull request by duduainankai**
Add support for freezer in cgroup v1 #9 .

### CODE REVIEW

**Comment by utam0k** [comment]
I'll review it tonight, so please wait.

**Comment by utam0k** [comment]
@duduainankai 
This is nice! Overall, it would be better if you added some more comments. In particular, I guessed this code was referring to the runc codes. It is too dificult for a first developer to see it. So please provide explanatory comments.

File: src/cgroups/v1/freezer.rs [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+                                log::debug!("frozen after {} retries", i)
+                            }
+                            return Ok(());
+                        }
+                        _ => {
+                            return Err(anyhow!("unexpected state {} while freezing", r.trim()));
+                        }
+                    }
+                }
+                return Err(anyhow!("unbale to freeze"));
```
Comment (utam0k) [inline src/cgroups/v1/freezer.rs:87]: You coud use `bail!`.

**Comment by utam0k** [comment]
It is nice idea. You may also want to include a brief summary if you possible.
Don't forget to make the link a permalink.
> How about just comment with the runc code link ?

File: src/cgroups/v1/freezer.rs (hunk 1 above)
Comment (duduainankai) [reply src/cgroups/v1/freezer.rs]: > It is nice idea. You may also want to include a brief summary if you possible.
> Don't forget to make the link a permalink.
> 
> > How about just comment with the runc code link ?

Updated. PATL.

**Comment by duduainankai** [comment]
> It is nice idea. You may also want to include a brief summary if you possible.
> Don't forget to make the link a permalink.
> 
> > How about just comment with the runc code link ?

Updated. PTAL.

**Comment by Furisto** [comment]
Sorry for delay. Will review in a few hours.

**Review by utam0k — APPROVED** [review]
LGTM
I'll leave the merging of this PR to @Furisto .

**Pull request merged**